src/praisonai-agents/praisonaiagents/agent/chat_mixin.py

```
@@ -549,41 +549,337 @@ def _process_stream_response(self, messages, temperature, start_time, formatted_
-                    except Exception as e:
-                        logging.warning(f"AFTER_COMPACTION hook failed: {e}")
-                        if getattr(self, '_strict_hooks', False):
-                            raise
-            except Exception as _ce:
-                if getattr(self, '_strict_hooks', False):
-                    raise
-                logging.debug(f"[compaction] skipped (non-fatal): {_ce}")
+        # --- Proactive Context Budget Management (default-on) ---
+        # Analyzes token budget BEFORE LLM call and applies appropriate strategy
+        try:
+            route, compacted_messages = self._compute_context_budget_and_route(
```

Ruff (0.15.15)

[warning] Unpacked variable `route` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)